distributed/scheduler.py

```diff
@@ -647,6 +647,10 @@ def add_worker(self, comm=None, address=None, keys=(), ncores=None,
                 self.has_what[address] = set()
                 self.worker_bytes[address] = 0
                 self.processing[address] = dict()
+                try:
+                    self.total_occupancy -= self.occupancy[address]
+                except KeyError:
+                    pass
                 self.occupancy[address] = 0
```

Comment:
Under what conditions would we get a `KeyError` here?

Reply:
It’s actually the usual case: when the worker is first added, this KeyError/pass will be taken; it’s only not a KeyError when workers are re-added (and possibly some other circumstances?)

Reply:
Ah I see, well then I'll change my question to, under what conditions would `address` be in `self.processing` but not in `self.occupancy`?  Should we perhaps avoid this situation altogether and consider it an error if one exists but not the other?

Reply:
I’m actually now wondering if it is possible to ever be anything but the KeyError path..

Reply:
I don't think the code guarantees that add_worker can never be called a second time for a worker, without the occupancy having been cleared first, so I think the try..except is necessary.

Reply:
We definitely call `add_worker` many times during a worker's lifetime (heartbeats call add_worker).  But the `address not in self.processing` should be enough to protect this part from repeated calls.  `self.processing[worker]` is created on first instantiation and only destroyed on `remove_worker`.

Reply:
Yes, I agree. The test is redundant; I'll remove it.